CONTRIBUTING.rst

```diff
@@ -51,7 +51,7 @@ Development Environment
 =======================
```

Comment:
The way I read this section is:

1. You need ZFS and Docker to run *all* the tests.
2. A quick way to do this is Vagrant.
3. On Ubuntu you also need these other packages.
4. You need these other things

There are two things confusing about this:
* Are the things in (4) - Python and `tox` Ubuntu only? Are they to be installed on the development machine even if using Vagrant?
* The first section is basically saying "use any platform including CentOS in Vagrant" then the second one is saying "this is for Ubuntu only" so I think I could use CentOS just fine without extra packages.

These can be solved by having a clear introduction of "The way to get an environment with all dependencies is to use Vagrant, to do this follow XXX", then a new heading for developing locally.

And relatedly:
* Do we need PyPy really? Buildbot doesn't have a PyPy builder (not new in this PR).
* To run all the tests we need more than just the packages mentioned here - ZFS, Docker, `libssl-dev`, `libffi-dev`, right? On CentOS we have all the packages in `flocker/vagrant/dev/bootstrap.sh` (though some of them are just for the release process) like `rpmdevtools`. I think that it is best to mention that some tests cannot be run on Ubuntu with this setup.

Perhaps some of the above is overly pedantic and we can assume a higher level of awareness from developers.